Libplanet/Blockchain/BlockChain.cs

```diff
@@ -682,31 +682,51 @@ public long GetNextTxNonce(Address address)
         /// <param name="currentTime">The <see cref="DateTimeOffset"/> when mining started.</param>
         /// <param name="append">Whether to <see cref="Append(Block{T}, StateCompleterSet{T}?)"/>
         /// the mined block.  Turned on by default.</param>
+        /// <param name="txBatchSize">A number to limit the count of transactions that will
+        /// be contained in the block.  <see cref="int.MaxValue"/> by default.</param>
         /// <param name="cancellationToken">
         /// A cancellation token used to propagate notification that this
         /// operation should be canceled.
         /// </param>
         /// <returns>An awaitable task with a <see cref="Block{T}"/> that is mined.</returns>
         /// <exception cref="OperationCanceledException">Thrown when
         /// <see cref="BlockChain{T}.Tip"/> is changed while mining.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when
+        /// <paramref name="txBatchSize"/> is negative.</exception>
         public async Task<Block<T>> MineBlock(
             Address miner,
             DateTimeOffset currentTime,
             bool append = true,
+            int txBatchSize = int.MaxValue,
             CancellationToken cancellationToken = default(CancellationToken)
         )
         {
+            if (txBatchSize < 0)
```

Comment:
I'm not sure to restrict the action that uses the chain `txBatchSize: 0`. How do you think about this cc @planetarium/libplanet

Reply:
I guess `txBatchSize` can be 0. If a user wants to include at least one tx in every blocks, may be he can restrict it at policy.

Reply:
At first, I thought it can be a matter that allowing `txBatchSize` because it can help to prevent [selfish-mining](https://github.com/planetarium/libplanet/issues/546). but on second thought, it can't help because miners still can discard any other txs by un-staging its all stated txs.